scylla-cql/src/serialize/row_tests.rs

```diff
@@ -1015,3 +1015,169 @@ fn ref_value_list() {
+    };
+
+    let reference = do_serialize(
+        (
+            &value.a,
+            &value.inner_one.x,
+            &value.inner_two.z,
+            &value.inner_one.y,
+        ),
+        &spec,
+    );
+
+    let row = do_serialize(value, &spec);
+
+    assert_eq!(reference, row);
+}
+
+#[test]
+fn test_flatten_row_serialization_with_enforced_order_and_skip_namecheck() {
+    #[derive(SerializeRow, Debug)]
+    #[scylla(crate = crate, flavor = "enforce_order")]
+    struct OuterColumns {
+        a: String,
+        #[scylla(flatten)]
+        inner_one: InnerColumnsOne,
+        d: i32,
+        #[scylla(flatten)]
+        inner_two: InnerColumnsTwo,
+    }
+
+    #[derive(SerializeRow, Debug)]
+    #[scylla(crate = crate, flavor = "enforce_order", skip_name_checks)]
+    struct InnerColumnsOne {
+        potato: bool,
+        carrot: f32,
+    }
+
+    #[derive(SerializeRow, Debug)]
+    #[scylla(crate = crate, flavor = "enforce_order")]
+    struct InnerColumnsTwo {
+        e: String,
+    }
+
+    let value = OuterColumns {
+        a: "A".to_owned(),
+        inner_one: InnerColumnsOne {
+            potato: false,
+            carrot: 2.3,
+        },
+        d: 32,
+        inner_two: InnerColumnsTwo { e: "E".to_owned() },
+    };
+
+    let spec = [
+        col("a", ColumnType::Native(NativeType::Text)),
+        col("b", ColumnType::Native(NativeType::Boolean)),
+        col("c", ColumnType::Native(NativeType::Float)),
+        col("d", ColumnType::Native(NativeType::Int)),
+        col("e", ColumnType::Native(NativeType::Text)),
+    ];
+
+    let reference = do_serialize(
+        (
+            &value.a,
+            &value.inner_one.potato,
+            &value.inner_one.carrot,
+            &value.d,
+            &value.inner_two.e,
+        ),
+        &spec,
+    );
+    let row = do_serialize(value, &spec);
+
+    assert_eq!(reference, row);
+}
+
+#[test]
+fn test_name_flatten_with_lifetimes() {
+    #[derive(SerializeRow)]
+    #[scylla(crate = crate, flavor = "enforce_order")]
+    struct Inner<'b> {
+        b: &'b bool,
+    }
+
+    #[derive(SerializeRow)]
+    #[scylla(crate = crate, flavor = "enforce_order")]
+    struct Outer<'a, 'b> {
+        #[scylla(flatten)]
+        inner: &'a Inner<'b>,
+    }
+
+    let b = true;
+    let inner = Inner { b: &b };
+
+    let value = Outer { inner: &inner };
+    let spec = [col("b", ColumnType::Native(NativeType::Boolean))];
+
+    let reference = do_serialize((&value.inner.b,), &spec);
+    let row = do_serialize(value, &spec);
+
+    assert_eq!(reference, row);
+}
+
+#[test]
+fn test_ordered_flatten_with_lifetimes() {
+    #[derive(SerializeRow)]
+    #[scylla(crate = crate, flavor = "enforce_order")]
+    struct Inner<'b> {
+        b: &'b bool,
+    }
+
+    #[derive(SerializeRow)]
+    #[scylla(crate = crate, flavor = "enforce_order")]
+    struct Outer<'a, 'b> {
+        #[scylla(flatten)]
+        inner: &'a Inner<'b>,
+    }
```

Comment:
:wrench: Just to make sure, let's also have a test case with `skip_name_checks` set.